signing/src/main/java/tech/pegasys/web3signer/signing/secp256k1/azure/AzureKeyVaultSigner.java

```diff
@@ -44,46 +44,39 @@ public class AzureKeyVaultSigner implements Signer {
   private final AzureConfig config;
   private final ECPublicKey publicKey;
   private final SignatureAlgorithm signingAlgo;
-  private final AzureKeyVaultFactory azureKeyVaultFactory;
   private final boolean needsToHash; // Apply Hash.sha3(data) before signing
+  private final AzureConnection azureConnection;
+  private final AzureKeyVault vault;
 
   AzureKeyVaultSigner(
       final AzureConfig config,
       final Bytes publicKey,
       final boolean needsToHash,
       final boolean useDeprecatedSignatureAlgorithm,
-      final AzureKeyVaultFactory azureKeyVaultFactory) {
+      final AzureKeyVault azureKeyVault,
+      final AzureConnection azureConnection) {
```

Comment:
Will having this as a strong reference, prevent the cache from disposing of the underlying HttpClient?

Reply:
Changed that along with some renames and some clean ups in the connection factory